File: lectures/14/Books/app/src/main/java/ro/ubbcluj/cs/books/EventDetailFragment.java
package ro.ubbcluj.cs.books;

import android.app.Activity;
import android.support.design.widget.CollapsingToolbarLayout;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.Date;

import ro.ubbcluj.cs.books.books.R;
import ro.ubbcluj.cs.books.domain.Book;

/**
 * A fragment representing a single Book detail screen.
 * This fragment is either contained in a {@link BookListActivity}
 * in two-pane mode (on tablets) or a {@link EventDetailActivity}
 * on handsets.
 */
public class EventDetailFragment extends Fragment {
  /**
   * The fragment argument representing the item ID that this fragment
   * represents.
   */
  public static final String ARG_ITEM_ID = "item_id";

  /**
   * The dummy content this fragment is presenting.
   */
  private Book mItem;

  /**
   * Mandatory empty constructor for the fragment manager to instantiate the
   * fragment (e.g. upon screen orientation changes).
   */
  public EventDetailFragment() {
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    if (getArguments().containsKey(ARG_ITEM_ID)) {
      // Load the dummy content specified by the fragment
      // arguments. In a real-world scenario, use a Loader
      // to load content from a content provider.
      mItem = new Book(Integer.parseInt(getArguments().getString(ARG_ITEM_ID)),"foo",new Date());

      Activity activity = this.getActivity();
      CollapsingToolbarLayout appBarLayout = (CollapsingToolbarLayout) activity.findViewById(R.id.toolbar_layout);
      if (appBarLayout != null) {
        appBarLayout.setTitle(mItem.getTitle());
      }
    }
  }

  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container,
                           Bundle savedInstanceState) {
    View rootView = inflater.inflate(R.layout.book_detail, container, false);

    // Show the dummy content as text in a TextView.
    if (mItem != null) {
      ((TextView) rootView.findViewById(R.id.event_detail)).setText(mItem.getTitle());
    }

    return rootView;
  }
}

File: lectures/14/Books/app/src/main/java/ro/ubbcluj/cs/books/EventDetailFragment.kt
package ro.ubbcluj.cs.books

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import ro.ubbcluj.cs.books.databinding.BookDetailBinding
import ro.ubbcluj.cs.books.domain.Book
import java.util.*


/**
 * A fragment representing a single Book detail screen.
 * This fragment is either contained in a [BookListActivity]
 * in two-pane mode (on tablets) or a [EventDetailActivity]
 * on handsets.
 */
/**
 * Mandatory empty constructor for the fragment manager to instantiate the
 * fragment (e.g. upon screen orientation changes).
 */
class EventDetailFragment : Fragment() {

  /**
   * The dummy content this fragment is presenting.
   */
  private lateinit var book: Book
  private lateinit var binding: BookDetailBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    if (arguments != null && arguments!!.containsKey(ARG_ITEM_ID)) {
      val bookId = arguments!!.getInt(ARG_ITEM_ID)
      val bookTitle = arguments!!.getString(ARG_ITEM_TITLE)
      book = Book(bookId, bookTitle, Date())
    }
  }

  override fun onCreateView(inflater: LayoutInflater,
                            container: ViewGroup?,
                            savedInstanceState: Bundle?): View? {
    binding = BookDetailBinding.inflate(inflater, container, false)
    val rootView = binding.root

    // Show the dummy content as text in a TextView.
    binding.eventDetail.text = book.title

    return rootView
  }

  companion object {
    /**
     * The fragment argument representing the item ID that this fragment
     * represents.
     */
    const val ARG_ITEM_ID = "item_id"
    const val ARG_ITEM_TITLE = "item_title"
  }
}
